network/wsPeer.go

```diff
@@ -525,3 +602,75 @@ func (wp *wsPeer) CheckSlowWritingPeer(now time.Time) bool {
 	timeSinceMessageCreated := now.Sub(time.Unix(0, ongoingMessageTime))
 	return timeSinceMessageCreated > maxMessageQueueDuration
 }
+
+// getNonce returns the byte representation of ever increasing uint64
+// The value is stored on wsPeer
+func (wp *wsPeer) getNonce() []byte {
+	atomic.AddUint64(&wp.nonce, 1)
```

Comment:
The function AddUint64 returns you a Uint64; use the returned value as the input to the binary.PutUvarint

Reply:
Rewrote the test and it failed with this implementation of getNonce()
This suggestion fixed getNonce()